### PR TITLE
Arduino as IDF component support in espidf builder

### LINKED ISSUE #24: Arduino.h not found - ESP32 with ESP-IDF Framework
Hi,
Copying issue from https://community.platformio.org/t/arduino-h-not-found-esp32-with-esp-idf-framework/1257 to platform-espressif32.

Currently I am using an ESP32 with the IDF Framework.
According to the docs https://github.com/espressif/arduino-esp3217 (section: Using as ESP-IDF component) the IDF Framework (supported by Platform IO and set up in the platformio.ino file) has an Arduino Core implemented and can be called by the ´initArduino()´ function. But compiler throws an error that Arduino.h cannot be found.

Has this been solved? if not, hope somebody can help here.

------------- Log Start -----------------
Processing esp32dev (platform: espressif32; lib_ldf_mode: deep; board: esp32dev; framework: espidf)
 
Verbose mode can be enabled via `-v, --verbose` option
Collected 0 compatible libraries
Looking for dependencies...
Project does not have dependencies
Compiling .pioenvs\esp32dev\src\main.o

Generating partitions .pioenvs\esp32dev\partitions_table.bin
Generating LD script .pioenvs\esp32dev\esp32_out.ld
Compiling .pioenvs\esp32dev\bootloader\bootloader_start.o
Compiling .pioenvs\esp32dev\bootloaderSupport\src\bootloader_flash.o
Compiling .pioenvs\esp32dev\bootloaderSupport\src\bootloader_random.o
Compiling .pioenvs\esp32dev\bootloaderSupport\src\efuse.o
Compiling .pioenvs\esp32dev\bootloaderSupport\src\esp_image_format.o
Compiling .pioenvs\esp32dev\bootloaderSupport\src\flash_encrypt.o
Compiling .pioenvs\esp32dev\bootloaderSupport\src\flash_partitions.o
src\main.c:8:52: fatal error: components/arduino/cores/esp32/Arduino.h: No such file or directory
compilation terminated.
*** [.pioenvs\esp32dev\src\main.o] Error 1
------------- Log End ------------------

Regards,
Andy

### PR DESCRIPTION
**Pull request by atanisoft**
Fixes #24.

By adding the following to the sdkconfig.h in the project using framework=espidf the framework-arduinoesp32 code is automatically included/compiled.

The sdkconfig.h is checked for CONFIG_AUTOSTART_ARDUINO and CONFIG_ENABLE_ARDUINO_DEPENDS which are defined by the arduino-esp32 Kconfig.

Also fixed an issue related to CONFIG_HEAP_POISONING_LIGHT leading to link errors (this is the default option from arduino-esp32)

This replaces PR #142.

### CODE REVIEW

File: builder/frameworks/espidf.py [hunk 1]
```diff
@@ -634,6 +644,79 @@ def build_espidf_bootloader():
         LINKFLAGS=["-T", "ulp_main.ld"]
     )
 
+if is_arduino_enabled(sdk_params):
+    ARDUINO_FRAMEWORK_DIR = platform.get_package_dir("framework-arduinoespressif32")
+    assert ARDUINO_FRAMEWORK_DIR and isdir(ARDUINO_FRAMEWORK_DIR)
+    print 'Including Arduino-ESP32 from %s' % ARDUINO_FRAMEWORK_DIR
+    env.Prepend(
```
Comment (ivankravets) [inline builder/frameworks/espidf.py]: Can we use default build script https://github.com/espressif/arduino-esp32/blob/master/tools/platformio-build.py ?

You can call it with https://github.com/platformio/platform-espressif32/blob/develop/builder/frameworks/arduino.py#L33

Comment (atanisoft) [reply builder/frameworks/espidf.py]: Unfortunately replacing the block with
```
SConscript(join(ARDUINO_FRAMEWORK_DIR, "tools", "platformio-build.py"))
```
results in:
```
*** Multiple ways to build the same target were specified for: C:\trains\platform-espressif32\examples\espidf-arduino-wifi\.pioenvs\esp32dev\partitions.bin  (from ['C:\\users\\mdunston\\.platformio\\packages\\framework-espidf\\components\\partition_table\\partitions_singleapp.csv'] and from ['partitions\\default.csv'])
File "C:\users\mdunston\.platformio\packages\framework-arduinoespressif32\tools\platformio-build.py", line 222, in <module>
```

Comment (ivankravets) [reply builder/frameworks/espidf.py]: @valeros please review

**Comment by atanisoft** [comment]
Closing as this is now available in the latest release, thanks @valeros!

**Pull request closed without merging**